### PR TITLE
Add a new tutorial which shows how to debug a failing pipeline

### PR DESCRIPTION
**Pull request by daniel-prefect**
Closes https://linear.app/prefect/issue/DOC-96/create-an-operator-tutorial-for-debugging-flow-runs

Preview: https://prefect-bd373955-debug_flow_runs_tutorial.mintlify.app/v3/tutorials/debug

### Checklist

- [X] This pull request references any related issue by including "closes `<link to issue>`"
- [ ] ~If this pull request adds new functionality, it includes unit tests that cover the changes~
- [ ] ~If this pull request removes docs files, it includes redirect settings in `mint.json`.~
- [ ] ~If this pull request adds functions or classes, it includes helpful docstrings.~

### CODE REVIEW

File: docs/v3/tutorials/debug.mdx [hunk 1]
```diff
@@ -0,0 +1,114 @@
+---
+title: Debug a data pipeline
+description: Find out when flow runs go wrong and how to troubleshoot them.
```
Comment (discdiver) [inline docs/v3/tutorials/debug.mdx:3]:
```suggestion
description: Learn how to troubleshoot flow runs that error.
```

File: docs/v3/tutorials/debug.mdx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    
+    return True
+
+# ...
+```
+
+Now, all flow runs succeed in spite of the `--fail-at-run` flag.
+Deploy the fix to the staging workspace to confirm this new behavior.
+
+```
```
Comment (discdiver) [inline docs/v3/tutorials/debug.mdx:91]:
```suggestion
```bash
```

File: docs/v3/tutorials/debug.mdx (hunk 1 above)
Comment (daniel-prefect) [reply docs/v3/tutorials/debug.mdx:3]: Error is a noun and I think a verb would be more grammatically correct here, so perhaps:

```suggestion
description: Learn how to troubleshoot flow runs that fail.
```

wdyt @discdiver?

File: docs/v3/tutorials/debug.mdx [hunk 3]
```diff
@@ -0,0 +1,114 @@
+---
+title: Debug a data pipeline
+description: Find out when flow runs go wrong and how to troubleshoot them.
+---
+
+In [Set up a platform for data pipelines](/v3/tutorials/platform), you used Prefect Cloud to set up a platform for data pipelines.
+In this tutorial, you'll learn what to do when those data pipelines fail.
+
+<Info>
+This tutorial starts where the [previous tutorial](/v3/tutorials/platform) leaves off, so complete that one first.
```
[the remark this answers is not in the document]
Comment (daniel-prefect) [reply docs/v3/tutorials/debug.mdx:10]: I would love to have a Terraform script they can run to provision Prefect Cloud automatically. I might need some help with this though.

File: docs/v3/tutorials/debug.mdx [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+Deploy the fix to the staging workspace to confirm this new behavior.
+
+```
+prefect cloud workspace --set "<account>/staging"
+python simulate_failures.py --fail-at-run 3
+```
+
+After the script finishes, open the **Home** page in Prefect Cloud to verify that the flow run is no longer failing.
+
+You can now deploy the fix to the production workspace as well.
```
[the remark this answers is not in the document]
Comment (daniel-prefect) [reply docs/v3/tutorials/debug.mdx:98]:
```suggestion
You can now switch workspaces to update the code used in the production workspace as well.
```

File: docs/v3/tutorials/debug.mdx (hunk 3 above)
[the remark this answers is not in the document]
Comment (discdiver) [reply docs/v3/tutorials/debug.mdx:10]: ooo. That would be cool. Could add as a linear issue.

**Review by discdiver — APPROVED** [review]
Cool.

File: docs/v3/tutorials/debug.mdx (hunk 3 above)
[the remark this answers is not in the document]
Comment (daniel-prefect) [reply docs/v3/tutorials/debug.mdx:10]: https://github.com/PrefectHQ/demos/pull/2

**Pull request merged**